core/src/main/java/net/mabako/steamgifts/adapters/viewholder/GiveawayListItemViewHolder.java

```diff
@@ -136,6 +140,35 @@ public void onError() {
         indicatorLevelNegative.setVisibility(giveaway.isLevelNegative() ? View.VISIBLE : View.GONE);
         indicatorPrivate.setVisibility(giveaway.isPrivate() ? View.VISIBLE : View.GONE);
         indicatorRegionRestricted.setVisibility(giveaway.isRegionRestricted() ? View.VISIBLE : View.GONE);
+
+        // Initialize the enter button
+        // Check if logged or the quick enter button setting is enabled
+        boolean loggedIn = SteamGiftsUserData.getCurrent(null).isLoggedIn();
+        if (!loggedIn || !PreferenceManager.getDefaultSharedPreferences(fragment.getContext()).getBoolean("preference_giveaway_show_quick_enter", false)){
+            giveawayEnterButton.setVisibility(View.GONE);
+        } else {
+            // Set the correct text based on the giveaway is already entered or not
+            if (giveaway.isEntered()) {
+                giveawayEnterButton.setText(activity.getString(R.string.leave_giveaway));
+                giveawayEnterButton.setEnabled(true);
+            } else {
+                giveawayEnterButton.setText(activity.getString(R.string.enter_giveaway));
+                //Check if the user have enough points, that the giveaway is not from him and that he have the required level
+                giveawayEnterButton.setEnabled(giveaway.getPoints() <= SteamGiftsUserData.getCurrent(null).getPoints() && giveaway.getLevel() <= SteamGiftsUserData.getCurrent(null).getLevel() && !SteamGiftsUserData.getCurrent(null).getName().equals(giveaway.getCreator()));
```

Comment:
How does the context menu determine that you can enter a giveaway? This looks like it should be a shared function.